### ISSUE TITLE
Video in classic block can prevent leaving the current page

### ISSUE THREAD

**Issue by barryceelen** [WordPress/gutenberg#10577]
Embedding a video in the classic editor can prevent the user from closing or leaving the current window. The 'are you sure you want to leave this site' popup is opened and subsequently closed automatically.

**To Reproduce**
1. Create a post with a classic editor block
2. Insert an embedded video, eg. `[embed]https://vimeo.com/172875996[/embed]`
4. Closing or navigating away from the current page without saving (sometimes) triggers the 'are you sure you want to leave this site' browser modal, and instantly closes that modal. This prevents navigating away from the current window.

The classic editor block seems to update the video embed before closing the window which may trigger the warning.

**Desktop (please complete the following information):**
 - OS: OSX
 - Browser: Chrome, Safari

**Comment by dijitul** [comment]
I actually got this "Are you sure.." popup error on many occasions, not added any videos.

**Comment by arunsathiya** [comment]
Noting that I was able to reproduce this behaviour as well on the `Classic` block. I tested using the same Vimeo link. 

My test results indicate:

- It prevents one from moving to another page site, until the post is auto-saved.
- When the auto-save kicks in, the navigation works as expected.
- When the auto-save does not kick in automatically, manual save using the button at the top right helps.
- This happens only for video embeds, and not for other embeds. 

[Video indicating this behaviour](https://cloudup.com/cDAoNQc6cpv). 

Console log on Google Chrome shows this.

```
JQMIGRATE: Migrate is installed, version 1.4.1
DevTools failed to parse SourceMap: https://gutenberg.app/wp-content/themes/boxy/admin/css/admin.css.map
DevTools failed to parse SourceMap: https://gutenberg.app/wp-includes/js/tinymce/skins/lightgray/skin.min.css.map
```

--- 

Tested on a local install of WordPress. WordPress 4.9.8 and Gutenberg 4.0.0 using Google Chrome Version 70.0.3538.67 (Official Build) (64-bit) on macOS 10.14 - Mac OS Mojave
